### PR TITLE
clang-8 warnings fix

### PR DESCRIPTION
**Pull request by egorovdanil**

### CODE REVIEW

**Comment by dvrogozh** [comment]
> We can remove Cmake flags from ASM if we explicitly specify the compiler for ASM as clang-8

How? Post example in commit message where you remove this flag.

File: tools/asg-hevc/include/frame_processor.h [hunk 1]
```diff
@@ -149,7 +149,14 @@ class FrameProcessor
     mfxU8 GetDefaultWeightedPredSample(mfxI32 predSampleLX);
     mfxU8 GetDefaultWeightedPredSample(mfxI32 predSampleL0, mfxI32 predSampleL1);
 
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wunused-private-field"
+#endif
```
Comment (dvrogozh) [inline tools/asg-hevc/include/frame_processor.h]: Why we just can't remove m_FrameType?

**Comment by egorovdanil** [comment]
I have added a `-DCMAKE_ASM_COMPILER=clang-8` flag to the command line.

File: tools/asg-hevc/include/frame_processor.h (hunk 1 above)
Comment (egorovdanil) [reply tools/asg-hevc/include/frame_processor.h]: I found a variable that was used with the same name, but this variable is of a different class. So now I have removed `m_FrameType`

Comment (dvrogozh) [reply tools/asg-hevc/include/frame_processor.h]: build fails...

**Comment by dvrogozh** [comment]
> I have added a -DCMAKE_ASM_COMPILER=clang-8 flag to the command line.

Add this to the _commit_ message, please where you modify CMakeLists.txt. That's a good catch by the way.

File: tools/asg-hevc/include/frame_processor.h [hunk 2]
```diff
@@ -149,8 +149,11 @@ class FrameProcessor
     mfxU8 GetDefaultWeightedPredSample(mfxI32 predSampleLX);
     mfxU8 GetDefaultWeightedPredSample(mfxI32 predSampleL0, mfxI32 predSampleL1);
 
+<<<<<<< Updated upstream
```
Comment (dvrogozh) [inline tools/asg-hevc/include/frame_processor.h]: git left garbage!

File: contrib/ipp/CMakeLists.txt [hunk 3]
```diff
@@ -46,7 +46,6 @@ endif()
 
 list( APPEND sources ${srcs})
 enable_language( C ASM )
-set( CMAKE_ASM_EXTENSIONS OFF )
```
Comment (egorovdanil) [inline contrib/ipp/CMakeLists.txt]: -DCMAKE_ASM_COMPILER=clang-8 flag delete CMake flags

File: samples/sample_plugins/rotate_opencl/src/sample_opencl_plugin.cpp [hunk 4]
```diff
@@ -72,20 +68,17 @@ mfxStatus Rotate::PluginInit(mfxCoreInterface *core)
     m_pmfxCore = new mfxCoreInterface;
     MSDK_CHECK_POINTER(m_pmfxCore, MFX_ERR_MEMORY_ALLOC);
     *m_pmfxCore = *core;
-    mfxCoreParam par = {0};
+    mfxCoreParam par = {{0}};
```
Comment (mgonchar) [inline samples/sample_plugins/rotate_opencl/src/sample_opencl_plugin.cpp]: just use empty initializer list:
```
 mfxCoreParam par = {};
```

Comment (egorovdanil) [reply samples/sample_plugins/rotate_opencl/src/sample_opencl_plugin.cpp]: done

File: contrib/ipp/CMakeLists.txt (hunk 3 above)
Comment (dvrogozh) [reply contrib/ipp/CMakeLists.txt]: I still don't see this in commit message.

Comment (egorovdanil) [reply contrib/ipp/CMakeLists.txt]: Now it's done

**Pull request merged**